string leg
string type
float64 angle